src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs

```diff
@@ -187,6 +206,60 @@ protected override IEnumerable<Model.AzureSqlManagedInstanceModel> GetEntity()
             return new List<AzureSqlManagedInstanceModel>() { ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name) };
         }
 
+        /// <summary>
+        /// Validate requested Hardware family.
+        /// </summary>
+        protected bool ShouldConfirmHardwareFamilyChange()
+        {
+            bool shouldConfirmHardwareFamilyChange = false;
+
+            ModelAdapter = InitModelAdapter();
+            AzureSqlManagedInstanceModel existingInstance = ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name);
+
+            // Get current hardware family
+            string currentHardwareFamily = existingInstance.Sku.Family;
+
+            // Get requested hardware family
+            string requestedHardwareFamily = !string.IsNullOrWhiteSpace(this.SkuName) ? AzureSqlManagedInstanceAdapter.GetHardwareGenerationFromSkuName(this.SkuName) : this.ComputeGeneration;
+
+            bool isHardwareFamilyChanged = !string.IsNullOrWhiteSpace(requestedHardwareFamily) && !currentHardwareFamily.Equals(requestedHardwareFamily, StringComparison.InvariantCultureIgnoreCase);
+
+            // Check whether hardware family is being changed to a deprecated hardware family
+            if (isHardwareFamilyChanged && requestedHardwareFamily.Equals(Constants.ComputeGenerationGen5, StringComparison.InvariantCultureIgnoreCase))
+            {
+                shouldConfirmHardwareFamilyChange = true;
+            }
+
+            return shouldConfirmHardwareFamilyChange;
+        }
+
+        /// <summary>
+        /// Get edition from SKU name.
+        /// </summary>
+        /// <returns>The model to send to the update</returns>
+        protected string GetEditionFromSkuName(string skuName)
+        {
+            if (string.IsNullOrWhiteSpace(skuName))
+            {
+                return null;
+            }
+
+            string edition = null;
+
+            if (skuName.Equals(Constants.GeneralPurposeGen4, StringComparison.InvariantCultureIgnoreCase)
+                || skuName.Equals(Constants.GeneralPurposeGen5, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Can you just check StartsWith("GP_") or StartsWith("BC_"), instead of having every combination programmed ?

Reply:
Even better would be if you implemented a function which is the opposite of GetInstanceSkuPrefix()

Reply:
Actually, this method is also redundant. I'll remove it